File: src/haplovis/schemas/file.py
from enum import IntEnum
from typing import List
from pydantic import BaseModel


class FileStatus(IntEnum):
    NO_FILE = 0
    INVALID = 1
    NEEDS_PRE_PROCESSING = 2
    PRE_PROCESSING = 3
    READY = 4


class FileIndex(IntEnum):
    GFA = 0
    PHENO = 1
    # GFF = *
    BOOKMARKS = 2


class File(BaseModel):
    id: int
    name: str = ""
    description: str
    status: FileStatus
    required: bool
    file_extensions: List[str]

File: src/models/file.ts
export interface UploadFile {
  readonly description: string;
  readonly status: UploadStatus;
  readonly required: boolean;
  readonly file_extensions: string[];
  readonly name?: string;
  readonly path?: string;
}

export enum UploadStatus {
  NO_UPLOAD = 0,
  SUCCESSFUL_UPLOAD = 1,
  WARNING_UPLOAD = 2,
}
